mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb

```diff
@@ -3347,11 +3347,7 @@ func testAccComputeInstance_nic_securityPolicyCreateWithTwoNicsAndTwoAccessConfi
 			},
 			{
 				Config: testAccComputeInstance_nic_securityPolicyCreateWithTwoNicsAndTwoAccessConfigsUpdateTwoPoliciesRemoveAccessConfig(suffix, policyName, policyName2, instanceName, "google_compute_region_security_policy.policyforinstance.self_link", "google_compute_region_security_policy.policyforinstance.self_link"),
-				Check: resource.ComposeTestCheckFunc(
-					testAccCheckComputeInstanceExists(
-						t, "google_compute_instance.foobar", &instance),
-					testAccCheckComputeInstanceNicAccessConfigHasSecurityPolicy(&instance, policyName),
-				),
+				ExpectError: regexp.MustCompile(errorDeleteAccessConfigWithSecPolicy),
```

Comment:
I'm a bit confused why we're expecting an error here. The tests are trying to update the resource google_compute_instance with only one change from
```
access_config {
  network_tier = "STANDARD"
}
security_policy = google_compute_region_security_policy.policyforinstance.self_link
```
to 
```
access_config {
  network_tier = "PREMIUM"
  nat_ip = google_compute_address.normal-address.address
}
security_policy = google_compute_region_security_policy.policyforinstance.self_link
```
Is this update not allowed?

Reply:
This update replaces current access config and the API throws an according error:
```
Error deleting old access_config: googleapi: Error 400: Invalid value for field 'accessConfig': 'external-nat'. Cannot delete an access config with a security policy set. Please remove the security policy first.
```

This is also described in API [documentation](https://cloud.google.com/compute/docs/reference/rest/v1/instances/setSecurityPolicy) of `instances.setSecurityPolicy`

>You can only set a security policy for network interfaces with an access config.

This is probably a new error in the API because in other tests we do this.
```
ExpectError: regexp.MustCompile("Error setting security policy to the instance since at least one access config must exist"),
```

The above error happens on creation of a `SecurityPolicy` but since we are trying to remove an access config that already has one set it fails from the `access_config` code